### PR TITLE
feat: partially functioning property behavior ui (no saving or loading yet)

### PR DESCRIPTION
**Pull request by EmosewaMC**
Updates the property behavior UI to be functional.

**Does not implement saving or loading of behaviors.**

Tested iteratively and at the end all messages in all (i believe all) contexts and did not spot any errors.  Fortunately, there are little pointers being accessed in behavior contexts outside of serialization, so any faults will be in vector accesses failing with their respective .ats.  The UI when you make changes and close the window should now save those changes on the local world for that behavior.  If you remove the behavior the changes will be lost.  If you shutdown the world the behavior will be lost.  If you pickup the model the behaviors will be lost.  This feature is not a small one and will take time for all the pieces to come together.

### CODE REVIEW

**Comment by EmosewaMC** [comment]
latest commit doesnt really make the include path smaller, but allows more work to be done in parallel

**Review by Xiphoseer — APPROVED** [review]

**Review by Xiphoseer — APPROVED** [review]

File: dGame/dPropertyBehaviors/State.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class AMFArrayValue;
+
+class State {
+public:
+	template<typename Msg>
+	void HandleMsg(Msg& msg);
+
+	void SendBehaviorBlocksToClient(AMFArrayValue& args) const;
+	bool IsEmpty() const;
```
Comment (aronwk-aaron) [inline dGame/dPropertyBehaviors/State.h:14]: Same here for operator overloading, wont point further instances out

**Review by aronwk-aaron — APPROVED** [review]
Approved with nits

File: dGame/dPropertyBehaviors/Strip.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+class AMFArrayValue;
+
+class Strip {
+public:
+	template<typename Msg>
+	void HandleMsg(Msg& msg);
+
+	void SendBehaviorBlocksToClient(AMFArrayValue& args) const;
+	bool IsEmpty() const { return m_Actions.empty(); }
```
[the remark this answers is not in the document]
Comment (aronwk-aaron) [reply dGame/dPropertyBehaviors/Strip.h:17]: cool

[the remark this answers is not in the document]
Comment (EmosewaMC) [reply dGame/dPropertyBehaviors/Strip.h:17]: Would you like the method to instead be called Empty?

[the remark this answers is not in the document]
Comment (aronwk-aaron) [reply dGame/dPropertyBehaviors/Strip.h:17]: it's fine as is

**Pull request merged**